theano/sandbox/gpuarray/conv.py

```diff
@@ -303,3 +310,266 @@ def c_code(self, node, nodename, inp, out_, sub):
         %(fail)s
     }
 """ % sub
+
+    def c_support_code_apply(self, node, name):
+        nb = 0
+        if self.kshp is not None:
+            nb = self.kshp[1]
+        kernels = self.gpu_kernels(node, name)
+        k = kernels[0]
+        code = """
+        #define THEANO_KERN_WID %(nb)d
+        """ % locals()
+        code += "\n".join([open(os.path.join(os.path.split(__file__)[0], f)).read()
+                           for f in ["conv_kernel.cu", "conv_full_kernel.cu"]])
+        kname = "conv_full_load_everything"
+        gk = gpuarray.GpuKernel(code, k.name, k.params, **k.flags)
+        bin = gk._binary
+        bcode = ','.join(hex(ord(c)) for c in bin)
+        code = code.replace('\\', '\\\\')
+        code = code.replace('"', '\\"')
+        code = code.replace('\n', '\\n')
+        mod = """
+        template <typename T>
+        static T ceil_intdiv(T a, T b)
+        {
+            return (a/b) + ((a %% b) ? 1: 0);
+        }
+        static const char conv_bcode[] = {%(bcode)s};
+        static const char *conv_code = "%(code)s";
+        """ % locals()
+        for k in kernels:
+            mod += "static GpuKernel " + k.name + '_' + name + ";\n"
+        mod += open(os.path.join(os.path.split(__file__)[0], "conv.cu")).read()
+        return mod
```

Comment:
Can you make an issue for this and tell in it that to clean up this tmp hack? Just to be sure we don't forget to update that part.